batch/batch/driver/instance_collection.py

```diff
@@ -133,17 +134,30 @@ async def check_on_instance(self, instance):
 
         # PROVISIONING, STAGING, RUNNING, STOPPING, TERMINATED
         gce_state = spec['status']
-        log.info(f'{instance} gce_state {gce_state}')
+
+        last_start_timestamp = spec.get('lastStartTimestamp')
+        if last_start_timestamp:
+            last_start_time_msecs = dateutil.parser.isoparse(last_start_timestamp).timestamp() * 1000
+        else:
+            last_start_time_msecs = None
+
+        log.info(f'{instance} gce_state {gce_state} last_start_timestamp {last_start_timestamp}')
+
+        if (gce_state == 'PROVISIONING'
```

Comment:
Do we want to kill any instance that is provisioning for more than 5 minutes?

Reply:
Yes, AFAICT, we pay for provisioning time.